### ISSUE TITLE
Feature Request: Add Resource for Elastic Cloud Integration

### ISSUE THREAD

**Issue by SignalRichard** [DataDog/terraform-provider-datadog#1933]
### Terraform Version

`1.4.6`

### Affected Resource(s)

Please list the resources as a list, for example:

-   datadog_integration_elastic_cloud

### Terraform Configuration Files

```hcl
resource "datadog_integration_elastic_cloud" {
  deployment_name = "deployment name"
  username = "username"
  password = "password"
  deployment_url = "https://..."
  tags = ["env:test"]
}
```

### Expected Behavior

It would be nice to have a resource that matched the global [Elastic Cloud integration](https://docs.datadoghq.com/integrations/elastic_cloud/) so that it could be provisioned alongside an Elastic Cloud deployment (which can also be done via Terraform) so that this part of the monitoring process does not need to be done manually.

### Actual Behavior

Currently this resource is not implemented.

**Comment by manan728** [comment]
Yes we would also like the Terraform resource for an more automated way of doing this integration, but shouldn't the resource be called `elastic_cloud_datadog_integration` instead since ec is the provider?

**Comment by stephenhand** [comment]
Would be great to see support for this - it would take an annoying manual step out of an otherwise fully automated deployment ;-)

**Comment by pellepelle3** [comment]
Any movement on this?
